### PR TITLE
WIP: JSON: Use structured unit IDs

### PR DESCRIPTION
**Pull request by nijel**
This allows to preserve structure of more complex documents and removes
difference between nested and non nested files.

This is proof of concept implementation for https://github.com/translate/translate/issues/3819, not ready for merge right now.

### CODE REVIEW

**Comment by nijel** [comment]
Caveats:

* Apparently it's not possible without changing convertors
* Anybody expecting unit id to be a string might need to update the code
* Currently it will break on unicode keys on Python 2, but that can be addressed if we decide to go this way

File: translate/convert/po2json.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def convertstore(self, includefuzzy=False, remove_untranslated=False):
         self.remove_untranslated = remove_untranslated
         self.inputstore.makeindex()
         for unit in self.templatestore.units:
-            inputunit = self.inputstore.locationindex.get(unit.getid())
+            inputunit = self.inputstore.locationindex.get(str(unit.getid()))
```
Comment (unho) [inline translate/convert/po2json.py]: Can't `unit.getid()` be altered to return a string?

Comment (nijel) [reply translate/convert/po2json.py]: That is certainly an option, this was mostly posted to get such kind of feedback - where to put the abstraction.

Comment (unho) [reply translate/convert/po2json.py]: Please note I've left a more elaborated comment on https://github.com/translate/translate/issues/3819#issuecomment-449043811

**Comment by nijel** [comment]
Closing in favor of https://github.com/translate/translate/pull/4091

**Comment by nijel** [comment]
Closing in favor of https://github.com/translate/translate/pull/4091

**Pull request closed without merging**